### PR TITLE
Removed deprecated export code

### LINKED ISSUE #4: Remove deprecated compatibility section
In `RequiredExports.h`, there is a `#ifndef NoDeprecated` section. This section was added a long time back during an API upgrade. It's meant to support existing level code that used old names that got renamed, and by providing automatic default level exports, rather than requiring a macro to be invoked to define them.

One of the reasons for the macro change, is for projects that want to use Outpost2DLL to link to Outpost2.exe functions, but are not level DLLs, and so don't need and don't want to define level data exports. By requiring them to invoke a macro to declare the exports, they are not automatically generated for all users of Outpost2DLL.

Meanwhile, level code can both declare and define the exports using the newer `ExportLevelDetails` and `ExportLevelDetailsEx` macros.

----

As this bit of legacy code is now years old, we should consider removing it. I'm not aware of any current level code that requires it.

### PR DESCRIPTION
**Pull request by DanRStevens**
This closes #4.

There's probably no need for this compatibility code anymore. Particularly not if we're using Git submodules that lock to a specific revision hash. If someone is upgrading to the newer version of Outpost2DLL, they can also upgrade their code to match.

### CODE REVIEW

File: RequiredExports.h [hunk 1]
```diff
@@ -77,41 +77,3 @@ struct BufferDesc
 		bufferDesc.bufferStart = 0; \
 		bufferDesc.length = 0; \
 	}
-
-
-// ** Deprecated **
-// ****************
-#ifndef NoDeprecated
-
-#define SCRIPT_API extern "C" __declspec(dllexport)
```
Comment (Brett208) [inline RequiredExports.h:86]: My two multiplayer scenarios use the SCRIPT_API to make the AI player work. I suspect removing this line will affect other scenarios as well.

```C++
SCRIPT_API SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

Is the solution to just write this instead?

```C++
extern "C" __declspec(dllexport) SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

I don't see any reason to keep this macro in service if this is all it is used for. Except that it is referenced a lot in tutorials and posts in the forums. If we want to remove it, how to add AI should probably be documented properly somewhere (Wiki maybe?).

@DanRStevens, thoughts?

-Brett

**Review by Brett208 — COMMENTED** [review]
If I remove the macro that I commented about and just use 'extern "C" __declspec(dllexport) instead, this compiles fine. The code comment should be fleshed out.

-Brett

File: RequiredExports.h (hunk 1 above)
Comment (DanRStevens) [reply RequiredExports.h:86]: When that section was deprecated, a renamed macro was added to replace it:
https://github.com/OutpostUniverse/Outpost2DLL/blob/72391d9cbe9c056a8d704ec2c2837ce36f75c6bb/RequiredExports.h#L8

I think the newer name is more clear about what it does.

Partially updated code would then be:
```cpp
Export SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

However, there was also the `ExportLevelDetails` and `ExportLevelDetailsEx` macros that were added, and which replace manually defining the DescBlock:
https://github.com/OutpostUniverse/Outpost2DLL/blob/72391d9cbe9c056a8d704ec2c2837ce36f75c6bb/RequiredExports.h#L49-L53

The equivalent code for newer levels is then something like:
```cpp
ExportLevelDetailsEx("6P, LoS, '<map name>'", "on6_01.map", "MULTITEK.TXT", MultiLastOneStanding, 6, 12, false)
```

Comment (DanRStevens) [reply RequiredExports.h:86]: One additional detail. The name `AIModDesc` is part of the exported API from Outpost2.exe. However, that type name doesn't appear in any method used by levels. Rather it appears in shell related code, which was split off into the `Outpost2App.h` header file. When the `Outpost2DLL` code was first written, we hadn't yet discovered the struct being exported by the levels matched up with the `AIModDesc` struct used by the shell methods. As it was a named data export (not a C++ function export), it didn't have any type information encoded in the name. Hence it ended up getting the made up name `SDescBlock`, which I believe came from the data exports variable name `DescBlock`. The `S` was prepended for "Struct", and to make type name unique from the exported variable name.

So basically all the old levels use a duplicate made up fake name of `SDescBlock`, rather than the proper name `AIModDesc`. That was part of the reason for the upgrade.

The other part was to remove the boiler plate of exporting the `MapName`, `LevelDesc` and `TechtreeName` fields, and also setting pointers to them in the `AIModDesc`. Since it's so formulaic, why bother to burden the level writer with the details.

**Comment by DanRStevens** [comment]
Merged master to resolve merge conflict from adding `const` on master branch to exports that are removed on this branch.

----

If many levels still use the old `SCRIPT_API` macro, perhaps we could keep that one macro and add a deprecated note that would appear when it is used.

Alternatively, we could consider if changing the macro from `SCRIPT_API` to `Export` is desirable. The current `LevelTemplate` project uses the newer `Export` macro.

**Comment by Brett208** [comment]
I'm using Export everywhere except when defining 

```C++
SCRIPT_API SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

For some reason it didn't occur to me that SCRIPT_API and Export were the same. I think removing SCRIPT_API is fine as it is an easy 1 for 1 swap with SCRIPT_API on any projects that may be using it.

I don't understand what you are saying AI multiplaer. How do you use ExportLevelDetailsEx to instantiate a legitimate AI player in a multiplayer scenario without using AiModDesc? 

Thanks,
Brett

**Comment by DanRStevens** [comment]
I don't understand that last part of what you wrote. The `ExportLevelDetailsEx` macro expands to include a definition of `AIModDesc DescBlock`. That's the last line of it.

The struct itself doesn't really instantiate any kind of AI player, though it does set the number of players, which is required to setup an AI player.

**Comment by DanRStevens** [comment]
Bump. I noticed this is still open.

I looked into this today. It seems I was confusing the `DescBlock` export and the `DescBlockEx` export. The two are distinct. The `ExportLevelDetails` and `ExportLevelDetailsEx` macros only deal with generating a `DescBlock`, not a `DescBlockEx`. Sorry for the confusion.

The solution to your problem here is simply replacing `SCRIPT_API` with `Export` (and possibly omitting zeros, as the compiler will auto initialize to zero for the remaining fields:
```
Export SDescBlockEx DescBlockEx = { 1 };
```

Perhaps we should consider updating the macros to account for the `DescBlockEx` export.

**Comment by Brett208** [comment]
Glad you took a second look at this. I was pretty confused. I'd like to merge now.

Thanks,
Brett
